modules/web-api.py

```diff
@@ -0,0 +1,253 @@
+            if resultsAPI.status_code != 200:
+                resultsAPI.raise_for_status()
+        #check for connection request 
+        except requests.exceptions.RequestException as e:
+            ProbeDescription['Connection'+reportName] = 'CRITICAL - Reports cannot connect to (https://' + arguments.hostname + pathToUse+'/' +reportName+'/'+reportTopologyGroup +'-'+ errmsg_from_excp(e)
+
+        #check for status code error  requests
+        except requests.exceptions.HTTPError as e:
+            ProbeDescription['Status'+reportName] = 'CRITICAL - Status code error. Cannot connect to (https://' + arguments.hostname + pathToUse+'/' +reportName+'/'+reportTopologyGroup +'-'+ errmsg_from_excp(e)
+
+
+        #when we have results report 
+        #check the json for the string we need
+        if (arguments.rtype=='ar'):
+            ProbeDescription[reportName] = 'OK - Availability for ' +reportName +' is OK'
+            try:
+                resultsjson['results'][0]['endpoints'][0]['results'][0]['availability']
+            except KeyError:
+                ProbeDescription[reportName] = 'CRITICAL - cannot retrieve availability from ' +reportName
+
+        #check status report
+        if (arguments.rtype=='status'):
+            ProbeDescription[reportName] = 'OK - status for ' +reportName +' is OK'
+            try:
+                resultsjson['groups'][0]['statuses']
+            except:
+                ProbeDescription[reportName] = 'CRITICAL - cannot retrieve status from ' +reportName
+
+    return ProbeDescription
+
+
+def debugValues(arguments):
+    """ Print debug values. 
+        Args:
+            arguments: the input arguments
+    """
+    if arguments.debug =='on':
+        print "[debugValues] - hostname:"+ arguments.hostname
+        print "[debugValues] - tenant:" + arguments.tenant
+        print "[debugValues] - rtype:" + arguments.rtype
+        print "[debugValues] - token:" + arguments.token
+        if arguments.timeout!='':
+            print "[debugValues] - timeout:" + str(arguments.timeout)
+        print "[debugValues] - debug:" + arguments.debug
+
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-H', dest='hostname', required=True, type=str, help='hostname')
+    parser.add_argument('-tenant', dest='tenant', required=True, type=str, default='EGI', help='tenant to check. Default EGI')
+    parser.add_argument('-rtype', dest='rtype', required=True, type=str, default='ar', help='status or ar (default ar)')
+    parser.add_argument('-token', dest='token', required=True, type=str, default='test_token', help='authentication token')
+    parser.add_argument('-day', dest='day', required=False, type=int, default=1, help='days to check (ex. 1 for yesterday, 2 for days ago) default yesterday')
+    parser.add_argument('-t', dest='timeout', required=False, type=int, default=180)
+    parser.add_argument('-d', dest='debug', required=False, type=str, default='off')
```

Comment:
``-d`` should be ``-v``, meaning verbose argument and it should be simple "flag" argument without any additional arguments. Check: https://nagios-plugins.org/doc/guidelines.html